src/main/java/com/google/api/codegen/discovery/DiscoveryProviderFactory.java

```diff
@@ -17,11 +17,15 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import com.google.api.Service;
 import com.google.api.codegen.ApiaryConfig;
+import java.io.File;
 import java.util.List;
 
 /** A factory for DiscoveryProviders which perform code generation. */
 public interface DiscoveryProviderFactory {
-  /** Create the provider from the given service, apiaryConfig, sampleConfigOverrides, and id. */
   DiscoveryProvider create(
-      Service service, ApiaryConfig apiaryConfig, List<JsonNode> sampleConfigOverrides, String id);
+      Service service,
+      ApiaryConfig apiaryConfig,
+      List<JsonNode> sampleConfigOverrides,
+      File rubyNamesFile,
```

Comment:
This seems like a non-language-specific file (as do some of the others in this PR), so why is there a parameter for Ruby and not for any other language?